### PR TITLE
lightning: always get latest PD leader when access PD after initialized

### LINKED ISSUE #43436: Lightning does not use auto-sync and fails when the PD cluster address changed
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

1. Start a TiDB cluster with 3 PDs ① ② ③ and a Lightning connected
2. Scale-out 3 more PDs ④ ⑤ ⑥
3. Wait 31 seconds
4. Scale-in the original PDs ① ② ③
5. Wait 31 seconds
6. Keep Lightning running.

### 2. What did you expect to see? (Required)

We make sure lightning succeed.

### 3. What did you see instead (Required)

Lightning will fail in some cases due to no valid pd-addr.

### 4. What is your TiDB version? (Required)

master

### LINKED ISSUE #46688: lightning failed at CreateOrUpdateRegionLabelRule when one PD node is unavailable
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

randomly inject some PD error to one node

### 2. What did you expect to see? (Required)

lightning import success

### 3. What did you see instead (Required)

```
Post "http://tc-a-pd:2379/pd/api/v1/config/region-label/rule": dial tcp 10.67.250.186:2379: connect: connection refused
github.com/pingcap/errors.AddStack
	/go/pkg/mod/github.com/pingcap/errors@v0.11.5-0.20220729040631-518f63d66278/errors.go:174
github.com/pingcap/errors.Trace
	/go/pkg/mod/github.com/pingcap/errors@v0.11.5-0.20220729040631-518f63d66278/juju_adaptor.go:15
github.com/pingcap/tidb/br/pkg/pdutil.pdRequestWithCode
	/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/br/br/pkg/pdutil/pd.go:202
github.com/pingcap/tidb/br/pkg/pdutil.pdRequest
	/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/br/br/pkg/pdutil/pd.go:153
github.com/pingcap/tidb/br/pkg/pdutil.(*PdController).CreateOrUpdateRegionLabelRule
	/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/br/br/pkg/pdutil/pd.go:948
github.com/pingcap/tidb/br/pkg/pdutil.(*PdController).pauseSchedulerByKeyRangeWithTTL
	/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/br/br/pkg/pdutil/pd.go:1010
github.com/pingcap/tidb/br/pkg/pdutil.(*PdController).PauseSchedulersByKeyRange
	/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/br/br/pkg/pdutil/pd.go:990
github.com/pingcap/tidb/br/pkg/lightning/backend/local.(*local).ImportEngine
	/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/br/br/pkg/lightning/backend/local/local.go:1645
github.com/pingcap/tidb/br/pkg/lightning/backend.(*ClosedEngine).Import
	/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/br/br/pkg/lightning/backend/backend.go:477
github.com/pingcap/tidb/br/pkg/lightning/restore.(*TableRestore).importKV
	/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/br/br/pkg/lightning/restore/table_restore.go:1039
github.com/pingcap/tidb/br/pkg/lightning/restore.(*TableRestore).importEngine
	/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/br/br/pkg/lightning/restore/table_restore.go:705
github.com/pingcap/tidb/br/pkg/lightning/restore.(*TableRestore).restoreEngines.func3
	/home/jenkins/agent/workspace/build-common/go/src/github.com/pingcap/br/br/pkg/lightning/restore/table_restore.go:333
runtime.goexit
	/usr/local/go/src/runtime/asm_amd64.s:1594
```

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve?

Issue Number: close #46688 close #43436

Problem Summary:

### What is changed and how it works?

PD nodes may change, so lightning should always try to use latest informtion rather than the addresses in task configuration.

### Check List

Tests 

- [ ] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: br/pkg/pdutil/pd.go [hunk 1]
```diff
@@ -329,6 +329,16 @@ func parseVersion(versionBytes []byte) *semver.Version {
 	return version
 }
 
+// TODO: always read latest PD nodes from PD client
+func (p *PdController) getAllPDAddrs() []string {
+	ret := make([]string, 0, len(p.addrs)+1)
+	if p.pdClient != nil {
+		ret = append(ret, p.pdClient.GetLeaderAddr())
```
Comment (lance6716) [inline br/pkg/pdutil/pd.go:336]: `PdController` is shared by BR, @3pointer PTAL

Comment (3pointer) [reply br/pkg/pdutil/pd.go:336]: LGTM

File: br/tests/lightning_pd_leader_switch/run.sh [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+wait $lightning_pid
+run_sql 'SELECT count(*), sum(c) FROM cpeng.a'
+check_contains 'count(*): 4'
+check_contains 'sum(c): 10'
+
+run_sql 'SELECT count(*), sum(c) FROM cpeng.b'
+check_contains 'count(*): 4'
+check_contains 'sum(c): 46'
+
+restart_services
```
Comment (D3Hunter) [inline br/tests/lightning_pd_leader_switch/run.sh:67]: what's this for?

**Review by D3Hunter — APPROVED** [review]

File: br/tests/lightning_pd_leader_switch/run.sh (hunk 2 above)
Comment (lance6716) [reply br/tests/lightning_pd_leader_switch/run.sh:67]: the old pd is killed, I'm afraid other tests will access the dead pd by address

**Review by Benjamin2037 — APPROVED** [review]
LGTM

**Comment by D3Hunter** [comment]
/hold
wait lightning test

**Comment by lance6716** [comment]
> /hold wait lightning test

lightning test passed before. You can see the commit hash is same
image
https://prow.tidb.net/?pull=46726

**Pull request merged**